forum/models/abstract.py

```diff
@@ -10,37 +10,49 @@
 class AbstractModel(ABC):
     """Abstract Class for Mongo model implementation"""
 
-    def __init__(self, collection_name):
-        db_name = 'cs_comments_service'
+    def __init__(self, collection_name, client=None):
         self.client = MongoBackend(collection=collection_name)
-        self.db = self.client[db_name]
-        self.collection = self.db[collection_name]
+        if client:
+            self.client = client
 
+        self.collection = self.client.collection
+
+    @property
     def get_collection(self):
+        """Get mongo db collection"""
         return self.collection
 
+    @property
     def get_db(self):
+        """Get mongo db instance."""
         return self.db
 
+    @property
     def get_client(self):
+        """Get mongo client"""
         return self.client
 
     @abstractmethod
-    def get(self, id):
+    def get(self, external_id):
+        """Get a document by ID"""
         pass
```

Comment:
Shouldn't this method (and all subsequent methods) raise `NotImplementedError`?

Reply:
yes, all future should have these implemented, added exception